src/platform/Darwin/PlatformManagerImpl.h

```diff
@@ -60,8 +54,20 @@ class PlatformManagerImpl final : public PlatformManager, public Internal::Gener
 
 private:
     // ===== Methods that implement the PlatformManager abstract interface.
-
-    CHIP_ERROR _InitChipStack(void);
+    CHIP_ERROR _InitChipStack();
+    CHIP_ERROR _Shutdown();
+
+    CHIP_ERROR _StartChipTimer(int64_t aMilliseconds) { return CHIP_ERROR_NOT_IMPLEMENTED; };
+    CHIP_ERROR _StartEventLoopTask() { return CHIP_NO_ERROR; };
+    void _RunEventLoop(){};
```

Comment:
So calling these functions is forbidden and a user shouldn't do it, right? Should they contain some logs or assert information about that? You could even consider catching calls to these functions at compile-time using the following technique:
```
#include <type_traits>

struct Base
{
    void f() {};
};

struct Derived : public Base
{
    template<class...>
    struct AlwaysFalse : std::false_type {};

    template<class... T>
    void f(T...) { static_assert(AlwaysFalse<T...>(), "Calling this function is forbidden"); }
};

void f()
{
    Base().f();
    // Derived().f(); <- this would generate a compiler error
}

Reply:
I can do that but so far I have avoided to change too much the code that relies on it for 2 reasons.

I tried to not have yet another #ifdef checking on which platform we are running for the code relying on the controller. To the best of my knowledge the only 2 platforms that relies on it from the controller point of view are `Linux` and `Darwin`. And I hope that we could get rid of the IO thread for `Linux` as well.

The second reason is that I was toying with the idea of starting a run loop for `chip-tool` instead of having it to rely on a https://github.com/project-chip/connectedhomeip/blob/bd7ced44b4a3233e1f5067fbcab3707a56f61fb6/examples/chip-tool/commands/common/Command.cpp#L379 and I was wondering I should reuse these methods.

Reply:
Ok, right, maybe rework of the CHIP main loop will get rid of the problem. It just seems a bit risky that some locking interface has an empty implementation on a certain platform, so a user must be aware of that.